girder/utility/mail_utils.py

```diff
@@ -28,6 +29,26 @@
 from girder.constants import SettingKey, PACKAGE_DIR
 
 
+def validateEmailAddress(address):
+    """
+    Determines whether a string is a valid email address.
+
+    This implements the grammar from 4.10.5.1.5 of the HTML Standard.
+
+    :param address: The string to test.
+    :type address: str
+
+    :rtype: bool
+    """
+    # https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address
+    emailRegex = re.compile(
+        r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+"
+        r"@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?"
+        r"(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
+    )
```

Comment:
Should we make this a module-level constant so it doesn't need to be recompiled each time this is called?

Reply:
I've just inlined everything into one function call. The method is called infrequently enough that there's no need to try and optimize it.